api/proto/teleport/accesslist/v1/accesslist_service.proto

```diff
@@ -340,22 +340,18 @@ enum PresetType {
 
 // Role describes a role to be upserted by Teleport for an access list.
 message Role {
-  // prefix_name is the prefix of the role name and allows the client to
+  // name_prefix is the prefix of the role name and allows the client to
   // control part of the role name so clients like the web UI can make some
   // assumptions based on this prefix e.g. what purpose the access role was
   // created for. For the final role name however, Teleport will add a suffix
-  // to this prefix_name before upserting the role. The suffix serves to:
+  // to this name_prefix before upserting the role. The suffix serves to:
   //  - ensure uniqueness by including the access list UID which also
   //    ties this role to that access list making it easier to query.
   //  - add a infix like "acl-preset" that easily identifies that this role
   //    belongs to an access list that was created using a preset.
-  string prefix_name = 1;
-  // spec defines the access to resources. If this is a nil spec,
-  // a role delete operation will be performed (if exists) and:
-  //  - if this role was assigned as grants, it will be removed
-  //    from access list
-  //  - if this role was assigned in another role (e.g. reviewer/requester)
-  //    it will be removed from that role
+  string name_prefix = 1;
+  // spec defines the access to resources. A nil spec will be interpreted as
+  // "no role update required".
```

Comment:
I think the cleanest design would be to always pass all the specs for all the roles in all requests and let the backend make the decisions. I don't think 1 extra role update for user initiated action is a problem here even if the spec doesn't change. And if it becomes a problem then we can start comparing specs in the backend, and skip the update if there are no changes.